lib/pages/AppConnectorConfig/AppConnectorConfig.tsx

```diff
@@ -0,0 +1,331 @@
+    appConnectorConfigDetails,
+    appConnectorSubscriptionDetails,
+    appConnectorSubscribed,
+  } = useSelector(appConnectorConfigSelector)
+
+  const dispatch = useDispatch()
+
+  const trans = useTranslation()
+
+  const t = (string: string, ...args) => {
+    return trans.t(`extensions.marketplace.${string}`, ...args)
+  }
+
+  const history = useHistory()
+  // 1. Subscription logic for the currently selected Marketplace app
+
+  // Retrieves the app's ID from the browser window's URL
+  const { appID } = useParams<any>()
+
+  const getSubscriptionStatus = () =>
+    appConnectorConfigDetails.data.workerStatus !== 'stopped' &&
+    appConnectorSubscribed
+
+  const backToApp = () => {
+    history.push(
+      `${encodeURI('/marketplace/app-details/' + selectedAppDetails.id)}`
+    )
+  }
+
+  const [fieldValues, setFieldValues] = useState({
+    apiUrl: '',
+    polling_toggle: getSubscriptionStatus(),
+  })
+  console.log('field values:', fieldValues)
+
+  const renderFieldsRaw = (entries) => {
+    return entries.map((entry, key) => (
+      <React.Fragment key={key}>
+        <Grid item xs={5}>
+          <TextField
+            label={t('appMarketplace.appConnectorConfig.appField')}
+            InputLabelProps={{
+              shrink: true,
+            }}
+            disabled={true}
+            style={{ width: 100 + '%' }}
+            value={entry}
+            variant="outlined"
+          />
+        </Grid>
+        <Grid item xs={2}>
+          <Icon
+            style={{
+              width: 100 + '%',
+              height: 100 + '%',
+              textAlign: 'center',
+              lineHeight: 50 + 'px',
+              color: '#62acee',
+            }}
+          >
+            arrow_right_alt
+          </Icon>
+        </Grid>
+        <Grid item xs={5}>
+          <TextField
+            label={t('appMarketplace.appConnectorConfig.apiField')}
+            InputLabelProps={{
+              shrink: true,
+            }}
+            style={{ width: 100 + '%' }}
+            name={entry}
+            value={fieldValues[entry]}
+            onChange={changeHandler}
+            variant="outlined"
+          />
+        </Grid>
+      </React.Fragment>
+    ))
+  }
+
+  const changeHandler = (changeEvent) => {
+    console.log(changeEvent)
+    const newFieldValues = { ...fieldValues }
+    newFieldValues[changeEvent.target.name] = changeEvent.target.value
+    setFieldValues(newFieldValues)
+  }
+
+  const updatePollingStatus = () => {
+    console.log('pooling toggle', fieldValues['polling_toggle'])
+  }
+
+  const saveSubscription = (event) => {
+    console.log('field values', fieldValues)
+    event.preventDefault()
+  }
+
+  const mappingIsOutdated = (currentMapping) => {
+    for (const currentMappingKey in currentMapping) {
+      if (
+        appConnectorConfigDetails.data.fieldsRaw.indexOf(currentMappingKey) ==
+        -1
+      )
+        return true
+    }
+    return false
+  }
+
+  useEffect(() => {
```

Comment:
This useEffect is very similar to the one below, the only difference being that one if statement calls one function, and the other if calls another one. That being said, perhaps we could reduce these two useEffects to a single one with both if statements! Small detail!